pkgs/development/python-modules/tuya-device-sharing-sdk/default.nix

```
@@ -0,0 +1,38 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, requests
+, paho-mqtt
+, cryptography
+}:
+let
+  pname = "tuya-device-sharing-sdk";
+  version = "0.2.0";
+in
+buildPythonPackage {
+  inherit pname version;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-fu8zh59wlnxtstNbNL8mIm10tiXy22oPbi6oUy5x8c8=";
+  };
+
+  postPatch = ''
+    touch requirements.txt
+  '';
+
+  doCheck = false;
```

```suggestion
  doCheck = false; # no tests
```

If the package does not have tests, we require configuring `pythonImportsCheck` with sensible modules that we can import to test the package doesn't break.

---

Actually there is [this](https://github.com/tuya/tuya-device-sharing-sdk/blob/9bfb454a7975e84145fac607cbc5060c77bc91c4/setup.py#L43) suite but I can't understand what it refers to.
With `doCheck = true` I get this:
```
python3.11-tuya-device-sharing-sdk> running test
python3.11-tuya-device-sharing-sdk> WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
python3.11-tuya-device-sharing-sdk> /nix/store/bvqadys5w6dallgim0y8y98h05b7kfn7-python3.11-setuptools-69.0.3/lib/python3.11/site-packages/setuptools/command/test.py:193: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated.
python3.11-tuya-device-sharing-sdk> !!
python3.11-tuya-device-sharing-sdk>         ********************************************************************************
python3.11-tuya-device-sharing-sdk>         Requirements should be satisfied by a PEP 517 installer.
python3.11-tuya-device-sharing-sdk>         If you are using pip, you can try `pip install --use-pep517`.
python3.11-tuya-device-sharing-sdk>         ********************************************************************************
python3.11-tuya-device-sharing-sdk> !!
python3.11-tuya-device-sharing-sdk>   ir_d = dist.fetch_build_eggs(dist.install_requires)
python3.11-tuya-device-sharing-sdk> /nix/store/bvqadys5w6dallgim0y8y98h05b7kfn7-python3.11-setuptools-69.0.3/lib/python3.11/site-packages/setuptools/command/test.py:194: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated.
python3.11-tuya-device-sharing-sdk> !!
python3.11-tuya-device-sharing-sdk>         ********************************************************************************
python3.11-tuya-device-sharing-sdk>         Requirements should be satisfied by a PEP 517 installer.
python3.11-tuya-device-sharing-sdk>         If you are using pip, you can try `pip install --use-pep517`.
python3.11-tuya-device-sharing-sdk>         ********************************************************************************
python3.11-tuya-device-sharing-sdk> !!
python3.11-tuya-device-sharing-sdk>   tr_d = dist.fetch_build_eggs(dist.tests_require or [])
python3.11-tuya-device-sharing-sdk> /nix/store/bvqadys5w6dallgim0y8y98h05b7kfn7-python3.11-setuptools-69.0.3/lib/python3.11/site-packages/setuptools/command/test.py:195: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated.
python3.11-tuya-device-sharing-sdk> !!
python3.11-tuya-device-sharing-sdk>         ********************************************************************************
python3.11-tuya-device-sharing-sdk>         Requirements should be satisfied by a PEP 517 installer.
python3.11-tuya-device-sharing-sdk>         If you are using pip, you can try `pip install --use-pep517`.
python3.11-tuya-device-sharing-sdk>         ********************************************************************************
python3.11-tuya-device-sharing-sdk> !!
python3.11-tuya-device-sharing-sdk>   er_d = dist.fetch_build_eggs(
python3.11-tuya-device-sharing-sdk> running egg_info
python3.11-tuya-device-sharing-sdk> writing tuya_device_sharing_sdk.egg-info/PKG-INFO
python3.11-tuya-device-sharing-sdk> writing dependency_links to tuya_device_sharing_sdk.egg-info/dependency_links.txt
python3.11-tuya-device-sharing-sdk> writing top-level names to tuya_device_sharing_sdk.egg-info/top_level.txt
python3.11-tuya-device-sharing-sdk> reading manifest file 'tuya_device_sharing_sdk.egg-info/SOURCES.txt'
python3.11-tuya-device-sharing-sdk> adding license file 'LICENSE'
python3.11-tuya-device-sharing-sdk> writing manifest file 'tuya_device_sharing_sdk.egg-info/SOURCES.txt'
python3.11-tuya-device-sharing-sdk> running build_ext
python3.11-tuya-device-sharing-sdk> runtests (unittest.loader._FailedTest.runtests) ... ERROR
python3.11-tuya-device-sharing-sdk> ======================================================================
python3.11-tuya-device-sharing-sdk> ERROR: runtests (unittest.loader._FailedTest.runtests)
python3.11-tuya-device-sharing-sdk> ----------------------------------------------------------------------
python3.11-tuya-device-sharing-sdk> ImportError: Failed to import test module: runtests
python3.11-tuya-device-sharing-sdk> Traceback (most recent call last):
python3.11-tuya-device-sharing-sdk>   File "/nix/store/3v2ch16fkl50i85n05h5ckss8pxx6836-python3-3.11.8/lib/python3.11/unittest/loader.py", line 162, in loadTestsFromName
python3.11-tuya-device-sharing-sdk>     module = __import__(module_name)
python3.11-tuya-device-sharing-sdk>              ^^^^^^^^^^^^^^^^^^^^^^^
python3.11-tuya-device-sharing-sdk> ModuleNotFoundError: No module named 'runtests'
python3.11-tuya-device-sharing-sdk> ----------------------------------------------------------------------
python3.11-tuya-device-sharing-sdk> Ran 1 test in 0.000s
python3.11-tuya-device-sharing-sdk> FAILED (errors=1)
python3.11-tuya-device-sharing-sdk> Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
python3.11-tuya-device-sharing-sdk> error: Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
error: builder for '/nix/store/wc5kpx8f7yvkwmz6dd7bilsjq8wipawi-python3.11-tuya-device-sharing-sdk-0.2.0.drv' failed with exit code 1;
```

So I simply disabled it.

---

I'm not adding `no tests` since I'm not sure that there are no tests, let me know if you want me to write something else.

---

I checked, there are none.

https://github.com/tuya/tuya-device-sharing-sdk

---

I'm adding `no tests` then, still wonder what `runtests.runtests` means...

---

Done, just rebased again.